resources/testharness.js

```diff
@@ -962,6 +962,10 @@ policies and contribution forms [3].
 
     function assert_array_equals(actual, expected, description)
     {
+        assert(actual && "length" in actual,
```

Comment:
This will throw if `actual` is not an object, no? (`in` throws if RHS is not an object)

I think we need `typeof actual === 'object' && actual !== null && 'length' in actual`.